go/oasis-test-runner/scenario/e2e/runtime/txsource.go

```diff
@@ -324,40 +331,90 @@ func (sc *txSourceImpl) manager(env *env.Env, errCh chan error) {
 	livenessTicker := time.NewTicker(sc.livenessCheckInterval)
 	defer livenessTicker.Stop()
 
+	if sc.nodeLongRestartDuration > 0 {
+		sc.Logger.Info("random long node restarts enabled",
+			"start_delay", sc.nodeLongRestartDuration,
+		)
+
+		// Setup long restart routine.
+		go func() {
+			for {
```

Comment:
This never seems to stop. Instead of using `context.Background` above, you could have a cancelable context where cancel gets called in a deferred handler. This way the goroutine would stop in case the test case shuts down.